website_mrp_bom/views/website_mrp_bom_templates.xml

```diff
@@ -0,0 +1,79 @@
+                </div>
+            <div class="oe_structure"/>
+        </div>
+    </t>
+</template>
+
+<template id="mrp_bom_detail_header" name="MRP BoM Details Header">
+        <h1 class="col-md-12 text-center" id="mrp_bom_name" t-field="bom.display_name"/>
+        <div class="col-md-4">
+            <t t-if="bom.product_tmpl_id">
+                <div t-field="bom.product_tmpl_id.image" t-options='{"widget": "image", "class": "center-block mb16", "max_width": 128}'/>
+            </t>
+            Base Quantity: <div t-field="bom.product_qty"/>
+            <div groups="product.group_uom" t-field="bom.product_uom_id.name"/>
+        </div>
+        <div class="col-md-8 mt32">
+            <t t-if="bom">
+                <div t-field="bom.website_description"/>
+            </t>
+        </div>
+</template>
+
+<template id="mrp_bom_detail_table" name="MRP BoM Details Table">
+    <div class="col-md-12">
+        <table class="table">
+             <thead>
+                <tr>
+                    <th>Image</th>
+                    <th>Product</th>
+                    <th>Quantity</th>
+                    <th>BoM</th>
+                </tr>
+            </thead>
+            <tbody>
+                <t t-foreach="bom.bom_line_ids" t-as="line">
+                    <tr style="font-weight: bold;">
+                        <td>
+                            <span t-field="line.product_id.sudo().image" t-options='{"widget": "image", "class": "center-block mb8", "max_width": 64}'/>
+                        </td>
+                        <td>
+                            <t t-if="line.product_id.sudo().product_tmpl_id.website_published">
+                                <a t-att-href="line.product_id.sudo().product_tmpl_id.website_url" style="display: block">
+                                    <span t-field="line.display_name"/>
+                                </a>
+                            </t>
+                            <t t-if="not line.product_id.sudo().product_tmpl_id.website_published">
+                                    <span t-field="line.display_name"/>
+                            </t>
+                        </td>
+                        <td>
+                            <span t-field="line.product_qty"/>
+                            <span groups="product.group_uom" t-field="line.product_uom_id.name"/>
+                        </td>
+                        <td>
+                            <t t-if="line.product_id.sudo().bom_ids and line.product_id.sudo().bom_ids[0].website_published">
+                                <a t-attf-href="/mrp_bom/#{ slug(line.product_id.bom_ids[0]) }" style="display: block">
+                                    <span class="fa fa-search"/>
+                                </a>
+                            </t>
+
+                        </td>
+                    </tr>
+                </t>
+            </tbody>
+        </table>
+    </div>
+</template>
+</odoo>
```

Comment:
I suggest if you can add a class in breakpoints so it would be easy to inherit the template.

It would be great if you can set something like below.

<t t-set="product_id t-value="line.product_id.sudo()"/> and reuse it in multiple place.